backend/src/main/java/com/team4/backend/service/InternshipOfferService.java

```diff
@@ -0,0 +1,30 @@
+package com.team4.backend.service;
+
+import com.team4.backend.dto.InternshipOfferDTO;
+import com.team4.backend.model.InternshipOffer;
+import com.team4.backend.model.Monitor;
+import com.team4.backend.repository.InternshipOfferRepository;
+import org.springframework.data.mongodb.core.aggregation.ArrayOperators;
+import org.springframework.stereotype.Service;
+import reactor.core.publisher.Mono;
+
+@Service
+public class InternshipOfferService {
+
+    private final InternshipOfferRepository internshipOfferRepository;
+
+    private final MonitorService monitorService;
+
+    public InternshipOfferService(InternshipOfferRepository internshipOfferRepository, MonitorService monitorService) {
+        this.internshipOfferRepository = internshipOfferRepository;
+        this.monitorService = monitorService;
+    }
+
+    public Mono<InternshipOfferDTO> addAnInternshipOffer(InternshipOfferDTO internshipOfferDTO) {
+
+        return monitorService.findMonitorByEmail(internshipOfferDTO.getEmailOfMonitor())
+                .flatMap(monitor -> internshipOfferRepository.save(new InternshipOffer(internshipOfferDTO, monitor)))
+                .map(InternshipOfferDTO::new);
+
```

Comment:
remove lines before and after return statement

Reply:
https://github.com/0xD34DC0DE/ProjetIntegreTeam4/pull/7/commits/96dfdd272ecb68a1945fec9f96a755fc0e8811ec